### PR TITLE
EVG-19918 fetch config all together

### PR DESCRIPTION
**Pull request by ybrill**
[EVG-19918](https://jira.mongodb.org/browse/EVG-19918)

### Description
We observed in the idle-host-termination job that each time we loaded the config it made ~30 round trips to the db and that caused the job to slow down. Although we aren't calling GetConfig there any more post #6790, getting them all in a single round trip to the db should make other calls to GetConfig much faster.

### Testing
This should be covered by the existing tests on GetConfig(), (i.e. everything in [the Admin suite](https://github.com/evergreen-ci/evergreen/blob/95c940e0db513c01e58f2ca106591dc289d2522c/config_test.go#L119))

### CODE REVIEW

**Comment by ybrill** [comment]
evergreen retry

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: config.go [hunk 1]
```diff
@@ -340,12 +335,6 @@ func BootstrapConfig(ctx context.Context) (*Settings, error) {
 			continue
 		}
 
-		// retrieve the section's document from the db
-		if err := section.Get(ctx); err != nil {
```
[the remark this answers is not in the document]
Comment (julianedwards) [reply config.go:344]: Ignore this comment, I see that sections are fetched individually so this needs to exist.

File: config_db.go [hunk 2]
```diff
@@ -124,6 +124,27 @@ func byId(id string) bson.M {
...
+}
+
+func getSectionsBSON(ctx context.Context, ids []string) ([]bson.Raw, error) {
+	cur, err := GetEnvironment().DB().Collection(ConfigCollection).Find(ctx, byIDs(ids))
+	if err != nil {
+		return nil, errors.Wrap(err, "finding configuration sections")
+	}
+
+	var docs = make([]bson.Raw, 0, len(ids))
+	for cur.Next(ctx) {
```
[the remark this answers is not in the document]
Comment (julianedwards) [reply config_db.go:138]: Ah I see, yeah you can't just fetch all the raw documents at once.

File: config_registry.go [hunk 3]
```diff
@@ -62,37 +67,67 @@ func resetRegistry() error {
...
+	}
+	idString, ok := id.StringValueOK()
+	if !ok {
+		return nil
+	}
+	section, ok := r.sections[idString]
+	if !ok {
+		return nil
+	}
+	return errors.Wrapf(bson.Unmarshal(rawSection, section), "unmarshaling section '%s'", idString)
```
[the remark this answers is not in the document]
Comment (ybrill) [reply config_registry.go]: done

File: config_registry.go [hunk 4]
```diff
@@ -19,8 +20,9 @@ import (
 var ConfigRegistry *ConfigSectionRegistry
```
[the remark this answers is not in the document]
Comment (ybrill) [reply config_registry.go]: Removed the config sections from global state.

**Review by julianedwards — APPROVED** [review]

**Pull request merged**